libr/anal/p/anal_arc.c

```diff
@@ -6,101 +6,955 @@
+                    op->type = R_ANAL_OP_TYPE_CALL;
+                }
+                op->jump = (addr & ~3) + limm;
+                op->fail = addr + op->size;
+            }
+            break;
+        case 2: { /* Load Register with Offset, 0x02 */
+            field_a = (words[0] & 0x0000003f);
+            field_b = (words[0] & 0x07000000) >> 24 | (words[0] & 0x7000) >>9;
+            imm     = sex_s9((words[0] & 0x00ff0000) >> 16 | (words[0] & 0x8000) >>7);
+            ut8 field_aa = (words[0] & 0x600) >> 9;
+            ut8 field_zz = (words[0] & 0x180) >> 7;
+
+            op->type = R_ANAL_OP_TYPE_LOAD;
+
+            switch(field_zz) {
+            case 0: op->refptr = 4; break;
+            case 1: op->refptr = 1; break;
+            case 2: op->refptr = 2; break;
+            default:
+                op->type = R_ANAL_OP_TYPE_ILL;
+                break;
+            }
+
+            if (field_b == 0x3e) {
+                op->size = 8;
+                limm = words[1];
+
+                switch (field_aa) {
+                case 0: /* No Field Syntax */
+                    op->ptr = limm+imm;
+                    break;
+                case 1: /* .A or .AW - invalid with limm */
+                case 2: /* .AB - invalid with limm */
+                    op->type = R_ANAL_OP_TYPE_ILL;
+                    break;
+                case 3: /* .AS */
+                    if (field_zz == 2) {
+                        op->ptr = limm+(imm<<1);
+                    } else if (field_zz == 0) {
+                        op->ptr = limm+(imm<<2);
+                    }
+                    break;
+                }
+            } else if (field_b == 0x3f) { /* PCL */
+                op->ptr = (addr & ~3) + imm;
+            }
+            break;
+        }
+        case 3: { /* Store Register with Offset, 0x03 */
+            field_c = (words[0] & 0xfc0) >> 6;
+            field_b = (words[0] & 0x07000000) >> 24 | (words[0] & 0x7000) >>9;
+            imm     = sex_s9((words[0] & 0x00ff0000) >> 16 | (words[0] & 0x8000) >>7);
+            /* ut8 field_aa = (words[0] & 0x18) >> 3; */
+            ut8 field_zz = (words[0] & 0x6) >> 1;
+
+            op->type = R_ANAL_OP_TYPE_STORE;
+
+            switch(field_zz) {
+            case 0: op->refptr = 4; break;
+            case 1: op->refptr = 1; break;
+            case 2: op->refptr = 2; break;
+            default:
+                op->type = R_ANAL_OP_TYPE_ILL;
+                break;
+            }
+
+            if (field_b == 0x3e) {
+                op->size = 8;
+                limm = words[1];
+                op->ptr = limm;
+            } else if (field_c == 0x3e) {
+                op->size = 8;
+                limm = words[1];
+                op->val = limm;
+            }
+
+            if (field_b == 0x3f) { /* PCL */
+                op->ptr = (addr & ~3) + imm;
+            }
+            break;
+        }
+        case 4: /* General Operations */
```

Comment:
I would consider splitting this case in another function, because it's quite a big `case` with many sub-switches and so.